### DISCUSSION TITLE
Getting more information out of minio for signature check failures

### DISCUSSION THREAD

**Discussion by io7m** [minio/minio#20283] [Q&A] [answered]
Hello!

I'm writing a small S3 client and I'm using `minio` as a local testbed. I'm running into issues where most requests succeed, but notably the `PutObject` action fails with a signature checking error. This suggests that something is wrong with my generation of the [canonical request](https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html#query-string-auth-v4-signing) when there's a payload.

According to the spec, in order to check a signature `minio` must be creating an equivalent canonical request internally to see if the signature it generates matches that of the one I'm sending. Is there some way to get more information out of `minio` so that I can see what canonical request it's creating?

**Comment by klauspost** [comment, marked as answer]
You can add printfs to the [code](https://github.com/minio/minio/blob/08d74819b66bf87aa270379c5cf374e0c9db8125/cmd/signature-v4.go#L347).

**Reply by io7m** [reply]
OK, thanks. It's not great, but it's better than nothing. At least it's all in one place.
